### PR TITLE
Call getCpeTypeForProtocol only if URL is not cached

### PR DESCRIPTION
**Pull request by theresa-m**
Previously getCpeTypeForProtocol was being called regardless of whether the URL was cached causing extra calls to j9file_attr.

Related to https://github.com/eclipse-openj9/openj9/pull/19380

### CODE REVIEW

File: runtime/jcl/j9jcl.tdf [hunk 1]
```diff
@@ -210,7 +210,7 @@ TraceEntry=Trc_JCL_com_ibm_oti_shared_getCachedToken_Entry Overhead=1 Level=1 Te
 TraceExit=Trc_JCL_com_ibm_oti_shared_getCachedToken_ExitFound Overhead=1 Level=1 Template="JCL: com.ibm.oti.shared getCachedToken: Exiting with found cached token"
 TraceExit=Trc_JCL_com_ibm_oti_shared_getCachedToken_ExitNull Overhead=1 Level=1 Template="JCL: com.ibm.oti.shared getCachedToken: Exiting with NULL"
 
-TraceEntry=Trc_JCL_com_ibm_oti_shared_getCachedURL_Entry Overhead=1 Level=2 Template="JCL: com.ibm.oti.shared getCachedURL: Entering for helperID=%d, cpeType=%d"
+TraceEntry=Trc_JCL_com_ibm_oti_shared_getCachedURL_Entry Overhead=1 Level=2 Template="JCL: com.ibm.oti.shared getCachedURL: Entering for helperID=%d"
```
Comment (pshipton) [inline runtime/jcl/j9jcl.tdf:213]: You can't change an existing tracepoint, a new one needs to be created, and this one can be marked Obsolete.

File: runtime/jcl/common/shared.c [hunk 2]
```diff
@@ -818,10 +828,6 @@ urlHashEqualFn(void* left, void* right, void *userData)
 		Trc_JCL_com_ibm_oti_shared_urlHashEqualFn_Exit1();
 		return 0;
 	}
-	if (leftItem->cpeType != rightItem->cpeType) {
-		Trc_JCL_com_ibm_oti_shared_urlHashEqualFn_Exit2();
```
Comment (pshipton) [inline runtime/jcl/common/shared.c:822]: Pls mark the tracepoint Obsolete if it's no longer used.

File: runtime/jcl/j9jcl.tdf (hunk 1 above)
Comment (pshipton) [reply runtime/jcl/j9jcl.tdf:213]: https://github.com/eclipse-openj9/openj9/blob/master/doc/diagnostics/AddingTracepoints.md

The idea is that you should be able to use the latest JVM to format a Snap file from any older version.

Comment (theresa-m) [reply runtime/jcl/j9jcl.tdf]: Thanks Peter. I updated the trace points.

File: runtime/jcl/common/shared.c [hunk 3]
```diff
@@ -71,7 +71,7 @@ typedef struct URLElements {
 } URLElements;
 
 
-static J9ClassPathEntry *getCachedURL(JNIEnv *env, jint helperID, const char *pathChars, jsize pathLen, UDATA cpeType, U_16 cpeStatus, const char *correctedPath);
+static J9ClassPathEntry *getCachedURL(JNIEnv *env, jint helperID, URLElements urlElements, U_16 cpeStatus);
```
[the remark this answers is not in the document]
Comment (theresa-m) [reply runtime/jcl/common/shared.c]: Thanks. I updated that.

**Comment by hangshao0** [comment]
@theresa-m There are merge conflicts.

**Comment by hangshao0** [comment]
Jenkins test sanity,extended win jdk8

**Comment by hangshao0** [comment]
Jenkins test sanity,extended amac jdk21

**Review by hangshao0 — APPROVED** [review]

**Pull request merged**